saleor/graphql/invoice/mutations.py

```diff
@@ -240,6 +259,12 @@ def perform_mutation(cls, _root, info, **data):
         )
         instance.status = JobStatus.SUCCESS
         instance.save(update_fields=["external_url", "number", "updated_at", "status"])
+        instance.order.events.create(
+            type=OrderEvents.INVOICE_GENERATED,
```

Comment:
Agree, it will look if you made a few edits to the invoice and in the history you would see a few entries that a new invoice was generated, can be confusing for dashboard users.